app/models/reservation.rb

```diff
@@ -2,15 +2,7 @@
 
 class Reservation < ApplicationRecord
   belongs_to :user
-  belongs_to :seat
+  has_many :seats
 
   validates_presence_of :user, :seat
-
-  def event
-    seat.event
-  end
-
-  def movie
-    event.movie
-  end
```

Comment:
You should be able to define these methods as associations through `seat`

has_one :event, through: :seat
has_one :movie, through: :event

^ this should work

Reply:
Done